### PR TITLE
[site/theme,#88]: Custom links component with support for video embeds in MDX

### PR DESCRIPTION
**Pull request by khalilcodes**
Video URL links such as https://www.youtube.com/watch?v=z2uAg-AIs-Y (Youtube, etc) surrounded by newlines in markdown will embed an iframe on the page.
  *Similar concept from: https://github.com/agentofuser/remark-oembed

Additional support for links added (since a custom link component is being created):
  * next/link for internal links (also checks for links containing a '.md' extension)
  * a tags for external links
 
Notes: moving to contentlayer, md pages now support raw html making the iframe display on page https://github.com/life-itself/web3/issues/88#issuecomment-1074792858

### CODE REVIEW

**Comment by rufuspollock** [comment]
@khalilcodes this looks like it has conflicts. Are you able to resolve and then we merge?

**Comment by khalilcodes** [comment]
@rufuspollock yes, will update the code.

**Comment by rufuspollock** [comment]
@khalilcodes is this ready to merge? Also can you point to an example page using it in the preview deploy ...

**Comment by khalilcodes** [comment]
@rufuspollock I will replace the iframe with the youtube link in one of the 'notes/...' page and commit so it can be previwed.

**Comment by khalilcodes** [comment]
@rufuspollock @sdiehl I would suggest we revert the concepts links and the ideologies folder after this merge since the links component has been added in this pr and would resolve #97. Also trimming the '.md' extensions support has been added for the next links, thus removing the need for page redirects if any.

Right now if we merge this the links are broken again as seen in preview.

**Comment by sdiehl** [comment]
I reverted the last commit that flattened the subdirectory.

**Comment by rufuspollock** [comment]
@khalilcodes can you summarize where we are at - i also note this PR has got a bit messy including both fixes to titles (?) and youtube change etc.

Can you put in a PR just for the titles fix. Then do a PR just for youtube stuff.

Aside: i think we create a "test.md" where we can put whatever we want to test and not mess with the other pages.

**Comment by khalilcodes** [comment]
Closing this PR and resubmitting a new one with youtube stuff only.
